### ISSUE TITLE
segfault when closing multiplayer window

### ISSUE THREAD

**Issue by CanadianJeff** [mgba-emu/mgba#180]
```
Program received signal SIGSEGV, Segmentation fault.
0x00000000004b6673 in GBASDLDetachPlayer (events=0x77a720 <QGBA::InputController::s_sdlEvents>, player=0xaaf0f0)
    at /build/mgba-vPUls_/mgba-0.4~WIP/src/platform/sdl/sdl-events.c:244
244	/build/mgba-vPUls_/mgba-0.4~WIP/src/platform/sdl/sdl-events.c: No such file or directory.
(gdb) bt full
#0  0x00000000004b6673 in GBASDLDetachPlayer (events=0x77a720 <QGBA::InputController::s_sdlEvents>, player=0xaaf0f0)
    at /build/mgba-vPUls_/mgba-0.4~WIP/src/platform/sdl/sdl-events.c:244
No locals.
#1  0x00000000004729c0 in QGBA::InputController::~InputController (this=0xaaf0b0, __in_chrg=<optimized out>)
    at /build/mgba-vPUls_/mgba-0.4~WIP/src/platform/qt/InputController.cpp:77
No locals.
#2  0x00000000004ab224 in QGBA::Window::~Window (this=0xaaf000, __in_chrg=<optimized out>)
    at /build/mgba-vPUls_/mgba-0.4~WIP/src/platform/qt/Window.cpp:158
No locals.
#3  0x00000000004ab2c9 in QGBA::Window::~Window (this=0xaaf000, __in_chrg=<optimized out>)
    at /build/mgba-vPUls_/mgba-0.4~WIP/src/platform/qt/Window.cpp:168
No locals.
#4  0x00007ffff5cda278 in QObject::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
No symbol table info available.
#5  0x00007ffff6e46efc in QWidget::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
No symbol table info available.
#6  0x00007ffff6f5813b in QMainWindow::event(QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
No symbol table info available.
#7  0x00007ffff6e0bc8c in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
```

**Comment by endrift** [comment]
I've seen this before but don't have a solid reproduction case yet; I'll look closer soon.

**Comment by endrift** [comment]
Fixed in 9590b484c73be4c852082a48912110695853b231.
